### DISCUSSION TITLE
Scrolling text

### DISCUSSION THREAD

**Discussion by paorie** [Alexays/Waybar#2006] [Q&A]
Is possible to use css animation for scrolling txt in custom module?
For example for a rss feeds reader.
tnks.

**Comment by Mato1111** [comment]
would love to have this for spotify song names that dont fit

**Comment by 5ysk3y** [comment]
This probably won't fit every use case but, should it help anyone, I solved scrolling text in Waybar for my music player using a couple of scripts, custom modules and [zscroll](https://github.com/noctuid/zscroll). zScroll itself hasnt been updated in a while (couple of years) but appears to remain functional.

Hopefully useful example:

In my use case, I have 2 scripts - one that contains commands to output the information I want to scroll on the bar, e.g:

`current_song.sh`
```
PLAYER_STATUS=$(playerctl -s -p cider status | tail -n1)
# Replace &'s in artist name/song title with +'s to avoid parsing issues
ARTIST=$(playerctl -p cider metadata artist | sed 's/&/+/g') 
TITLE=$(playerctl -p cider metadata title | sed 's/&/+/g')

if [[ $PLAYER_STATUS == "Paused" || $PLAYER_STATUS == "Playing" ]]; then
    echo "${ARTIST} - ${TITLE}"
elif [[ $PLAYER_STATUS == "Stopped" ]]; then
    echo "No Music Playing"
else
   echo "Music Player Offline"
fi
```

So this basically just outputs `Artist Name - Song title` if the defined player reports as online.

Then I pass this script into a second script for zscroll:

`music_panel.sh`
```
CURRENT_SONG="PATH/TO/SCRIPTS/current_song.sh"

zscroll -p " | " --delay 0.2 \
    --length 40 \
    --match-command "playerctl -p cider status" \
    --match-text "Playing" "--scroll 1" \
    --match-text "Paused" "--before-text ' ' --scroll 0" \
    --update-interval 1 \
    --update-check true $CURRENT_SONG &
wait
```

This will cause text to scroll when the combined artist/title name exceed the defined length value (e.g. 40 chars), and itll only scroll if the output of `match-command` returns `playing`. If it is `paused`, it will stop scrolling completely and whack a pause symbol before the output to denote that media is paused in Waybar. That then loops around the current song script, and in my waybar config I have this:

```
    "custom/media": {
      "escape": "true",
      "exec": "/PATH/TO/SCRIPTS/music_panel.sh",
      "format": "<U+F001> {}", 
      "max-length": 40,
      "on-click": "playerctl play-pause",
      "on-click-right": "playerctl stop",
      "on-scroll-down": "playerctl previous",
      "on-scroll-up": "playerctl next"
    },
```

Along with some CSS that styles it :)

**Reply by gaganchandan** [reply]
@5ysk3y I tried this, but it doesn't seem to be working. The output of  `music_panel.sh` is correct in the terminal, but no output is shown on the bar itself. What could be the issue?

**Reply by 5ysk3y** [reply]
Hey! So this could be anything really.. if you're able to run the scripts independently via a terminal successfully then Waybar *should* execute them in the same way (assuming you're running Waybar as the same user).

If you run Waybar directly from a terminal so you can get the stdout/stderr stuff, is there anything in the output that shows a potential issue I wonder?

**Reply by gaganchandan** [reply]
Sorry about the late response. I tried this and it doesn't seem to output any error messages.

**Reply by lxvevery1** [reply]
@gaganchandan try to remove interval field in waybar config!

**Comment by citr0net** [comment]
I know this is a dead topic, but may I suggest this repo? https://github.com/sicista33/waybar-scrolling-mpris

**Comment by BEST8OY** [comment]
I made a rust MPRIS module for my use case with AI
It's still a bit raw, might add more stuff in future as AI progresses :) (It can already do much more if I get the correct prompt to ask of it)
https://github.com/BEST8OY/ScrollMPRIS/
![image](https://github.com/user-attachments/assets/4841ec5a-8b9e-440c-93f2-3f4c84ac72b0)
![image](https://github.com/user-attachments/assets/e10fc029-d058-486c-9d93-1171d8d35d13)
